plugins/workflow/category/forms/action.xml

```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
+<form>
+    <fields name="options">
+        <fieldset name="actions"
+            label="COM_WORKFLOW_TRANSITION_ACTIONS_LABEL"
+        >
+            <field
+                name="category_id"
+                type="category"
+                label="PLG_WORKFLOW_CATEGORY_FIELD_CATEGORY_LABEL"
+                description="PLG_WORKFLOW_CATEGORY_FIELD_CATEGORY_DESC"
+                extension="com_content"
+            />
```

Comment:
Make sure, to have the posibility to not select a category at all.

Also workflow is more than "com_content", there could be 3rd party extension using categories + workflow, so there is the need for a dynamic way for the "extension".